### PR TITLE
User guide freshening...

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR involves a proofread pass through the RAJA user guide, except for the tutorial sections, and fixes formatting issues, typos, updates documentation in places, etc.  The generated sphinx docs for this PR can be found here: https://raja.readthedocs.io/en/bugfix-ug-fixes/

### CODE REVIEW

File: docs/sphinx/user_guide/index.rst [hunk 1]
```diff
@@ -65,11 +59,27 @@ RAJA helps developers insulate application loop kernels from underlying
...
 (loop bodies) and C++ templates (loop execution methods). This approach 
-promotes the perspective that developers should focus on tuning 
+promotes the perspective that application developers should focus on tuning 
 loop patterns rather than individual loops as much as possible. RAJA makes it 
 relatively straightforward to parameterize an application using execution 
 policy types so that it can be compiled in a specific configuration suitable 
-to a given architecture. 
+to a given architecture.
+
+RAJA support for various exexution back-ends is the result of collaborative
```
Comment (artv3) [inline docs/sphinx/user_guide/index.rst:68]: exexution -> execution

Comment (rhornung67) [reply docs/sphinx/user_guide/index.rst]: Thanks. Fixed.

File: docs/sphinx/user_guide/feature/reduction.rst [hunk 2]
```diff
@@ -53,8 +56,8 @@ RAJA supports five common reduction types:
 .. note:: * When ``RAJA::ReduceMinLoc`` and ``RAJA::ReduceMaxLoc`` are used 
             in a sequential execution context, the loop index of the 
             min/max is the first index where the min/max occurs.
-          * When the 'loc' reductions are used in a parallel execution context, 
-            the loop index given for the reduction value may be any index 
+          * When theese reductions are used in a parallel execution context,
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/feature/reduction.rst:59]: typo

File: docs/sphinx/user_guide/feature/view.rst [hunk 3]
```diff
@@ -164,9 +166,15 @@ it using indices in :math:`[-5, 5]`. In other words, one can use the loop::
   } 
 
 to initialize the values of the array. Each 'i' loop index value is converted
-to array offset access index by subtracting the lower offset to it; i.e., in 
+to an array offset indexx by subtracting the lower offset from it; i.e., in
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/feature/view.rst:169]: typo

File: docs/sphinx/user_guide/feature/view.rst [hunk 4]
```diff
@@ -210,14 +213,18 @@ tutorial sections.
...
+
 Typed Layouts
 ^^^^^^^^^^^^^
 
 RAJA provides typed variants of ``RAJA::Layout`` and ``RAJA::OffsetLayout``
-enabling user specified index types. Basic usage requires specifying types for
-the linear index, and the multi-dimensional indicies. The following example creates
-typed layouts wherein the linear index is of type TIL and the multidimensional
-indices are TIX, TIY,::
+enabling users to  specify integral index types. Basic usage requires
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/feature/view.rst:224]: typo

File: docs/sphinx/user_guide/feature/view.rst [hunk 5]
```diff
@@ -321,16 +334,16 @@ way to do this in parallel using OpenMP and a RAJA atomic view::
...
-runtime bounds checking for RAJA Views. This may be a useful debugging aid for
-users. When bounds checkoing is turned off (default case), there is no 
-additional run time overhead incurred. Bounds checking is accomplished within
-RAJA layouts (both offset and standard layouts). Upon an out of bounds error, 
+runtime bounds checking for RAJA views. This may be a useful debugging aid for
+users. When attempting to use an index value that is out of bounds,
 RAJA will abort the program and print the index that is out of bounds as
-well the value of the index and bounds.
+well the value of the index and bounds. Since the bounds checking is a runtime
+operation, it incurs non-negligible overhead. When bounds checkoing is turned
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/feature/view.rst:348]: typo

File: docs/sphinx/user_guide/tutorial.rst [hunk 6]
```diff
@@ -217,26 +221,26 @@ with respect to RAJA usage. We describe them here.
...
 directory. Additional information about the RAJA features 
 used can be found in :ref:`features-label`.
 
 The examples demonstrate CPU execution (sequential, SIMD, OpenMP
 multi-threading) and CUDA GPU execution. Examples that show how to use
 RAJA with other parallel programming model back-ends that are in 
-development will appear when we feel RAJA support for them is sufficiently
-complete and robust. For adventurous users who wish to try experimental 
-features, usage is similar to what is shown in the examples here.
+development will appear in furture RAJA releases. For adventurous users who
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/tutorial.rst:241]: typo

File: docs/sphinx/user_guide/feature/reduction.rst (hunk 2 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/feature/reduction.rst]: Fixed.

File: docs/sphinx/user_guide/feature/view.rst (hunk 3 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/feature/view.rst]: Fixed.

File: docs/sphinx/user_guide/feature/view.rst (hunk 4 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/feature/view.rst]: Fixed.

File: docs/sphinx/user_guide/feature/view.rst (hunk 5 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/feature/view.rst:349]: Fixed.

File: docs/sphinx/user_guide/tutorial.rst (hunk 6 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/tutorial.rst]: Fixed.

**Pull request merged**